src/diffusers/image_processor.py

```diff
@@ -882,3 +883,56 @@ def preprocess(
             depth = self.binarize(depth)
 
         return rgb, depth
+
+
+class IPAdapterMaskProcessor(VaeImageProcessor):
+    """
+    Image processor for IP Adapter image masks.
+
+    """
+    def __init__(self):
```

Comment:
I think the resize related arguments are really useful here too, in case you need to resize your mask! 
```suggestion
    def __init__(self):
        do_resize: bool = True,
        vae_scale_factor: int = 8,
        resample: str = "lanczos",
        do_normalize: bool = False,
        do_binarize: bool = True,
        do_convert_grayscale: bool = True)
```

Reply:
I don't fully understand why it is needed `to_resize` here.
Is it not sufficient to perform the resize by interpolation in the `downsample` method?